### PR TITLE
Add test that reproduces Runtime SDE bug.

### PR DESCRIPTION
**Pull request by mbeckerle**
DAFFODIL-2629

### CODE REVIEW

File: daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/setVarWIthValueLength.tdml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      <complexType>
+        <sequence>
+          <element name="someElement" type="xs:int"/>
+          <!--
+          DAFFODIL-2629 This setVariable calling dfdl:valueLength on the prior element. Causes Runtime SDE."
+          -->
+          <sequence>
+            <annotation>
+              <appinfo source="http://www.ogf.org/dfdl/">
+                <dfdl:setVariable ref="f:someVar">{ dfdl:valueLength(someElement, "bytes") }
```
Comment (stevedlawrence) [inline daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/setVarWIthValueLength.tdml:51]: If you use the `value` attribute for the expression does it still fail? Might be good to add a test for both syntaxes to make sure however this is resolved works for both.

File: daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/setVarWIthValueLength.tdml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                </dfdl:setVariable>
+              </appinfo>
+            </annotation>
+          </sequence>
```
Comment (stevedlawrence) [inline daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/setVarWIthValueLength.tdml:55]: Is it worth added an inputValueCalc and/or outputValueCalc to ensure that the variable actually gets the right value? I could imagine a but where the SDE goes away but doesn't actually calculate the correct value and we wouldn't know.

**Review by stevedlawrence — APPROVED** [review]
+1

File: daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/setVarWIthValueLength.tdml (hunk 1 above)
Comment (mbeckerle) [reply daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/setVarWIthValueLength.tdml]: Yup. I  added a test to verify this.

File: daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/setVarWIthValueLength.tdml (hunk 2 above)
Comment (mbeckerle) [reply daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/setVarWIthValueLength.tdml:54]: Added a subsequent assertion that the variable gets value 4.

**Pull request merged**